Detectors/TRD/simulation/src/Digitizer.cxx

```diff
@@ -54,18 +52,20 @@ Digitizer::Digitizer()
   mSDigits = false;
 }
 
-void Digitizer::process(std::vector<HitType> const& hits, std::vector<Digit>& digits)
+void Digitizer::process(std::vector<HitType> const& hits, DigitContainer_t& digitCont)
 {
   // (WIP) Implementation for digitization
 
   // Check if Geometry and if CCDB are available as they will be requiered
   // const int nTimeBins = mCalib->GetNumberOfTimeBinsDCS(); PLEASE FIX ME when CCDB is ready
 
+  ArrayADCMapContainer_t adcMapCont;
+
   // Loop over all TRD detectors
   // Get the a hit container for all the hits in a given detector then call convertHits for a given detector (0 - 539)
+  std::array<std::vector<HitType>, kNdet> hitsPerDetector;
+  getHitContainerPerDetector(hits, hitsPerDetector);
```

Comment:
It seems to be inefficient to copy the hits of each detector into a separate vector. It would be simpler to get iterators to the start of the hits for each detector, and then loop between the iterators for each chamber. 
Or loop over all hits in one go.

Reply:
This step can still be improved, as we talked using a vector of iterators. Maybe we can keep this as a separate improvement.